docs: Add 1.4.0 stable release notes (#6260)
spiceai/spiceai: docs/release_notes/v1.4.0.md

See the [Arrow 55.0.0 Changelog](https://github.com/apache/arrow-rs/blob/55.0.0/CHANGELOG.md) and [Arrow 55.1.0 Changelog](https://github.com/apache/arrow-rs/blob/55.1.0/CHANGELOG.md) for details.

### Runtime Highlights

**Search Result Caching:** Spice now supports runtime caching for search results, improving performance for subsequent searches and chat completion requests that use the `document_similarity` LLM tool. Caching is configurable with options like maximum size, item TTL, eviction policy, and hashing algorithm.

Example `spicepod.yml` configuration:

```yaml
runtime:
  caching:
    search_results:
      enabled: true
      max_size: 128mb
      item_ttl: 5s
      eviction_policy: lru
      hashing_algorithm: siphash
```

For more information, refer to the [Caching documentation](https://spiceai.org/docs/features/caching).

**AWS Glue Catalog Connector Alpha**: Connect to AWS Glue Data Catalogs to query Iceberg, Parquet, or CSV tables in S3.

Example `spicepod.yml` configuration:

```yaml
catalogs:

>   - from: glue

Reviewer: ```suggestion
  - from: glue:my_database
```

>   - from: glue:my_database

Reviewer: ```suggestion
  - from: glue
```

Example `spicepod.yml` configuration:

```yaml
datasets:
  - name: my_dataset
    from: s3://my-bucket/my_file.parquet
    acceleration:
      refresh_cron: 0 0 * * * # Daily refresh at midnight
```

For more information, refer to the [Dataset Refresh Schedules documentation](https://spiceai.org/docs/features/data-acceleration/data-refresh#refresh-schedules).

**Worker Execution Schedules:** Workers now support a `cron` parameter and will execute an LLM-prompt or SQL query automatically on the defined cron schedule, in conjunction with a provided `params.prompt`.

Example `spicepod.yml` configuration:

```yaml
workers:
  - name: email_reporter
    models:
      - from: gpt-4o
    params:
      prompt: 'Inspect the latest emails, and generate a summary report for them. Post the summary report to the connected Teams channel'
    cron: 0 2 * * * # Daily at 2am
```

For more information, refer to the [Worker Execution Schedules documentation](https://spiceai.org/docs/reference/spicepod/workers#cron).

> **SQL Worker Actions:** Spice now supports workers with `sql` actions, to execute automated SQL queries on a cron schedule:

Reviewer: ```suggestion
**SQL Worker Actions:** Spice now supports workers with `sql` actions for automated SQL query execution on a cron schedule:
```
